tracing/tracing.go

```diff
@@ -25,10 +26,56 @@ func InstrumentHTTP(h http.Handler) http.Handler {
 		}
 
 		ctx := req.Context()
+		ctx = CtxWithTraceID(ctx, traceid)
+
 		log := slog.FromCtx(ctx)
 		log = log.With("trace_id", traceid)
 		ctx = slog.NewContext(ctx, log)
 
 		h.ServeHTTP(res, req.WithContext(ctx))
 	})
 }
+
+// CtxWithTraceID creates a new [context.Context] with the given trace ID associated with it.
+// Call [CtxGetTraceID] to retrieve the trace ID.
+func CtxWithTraceID(ctx context.Context, traceID string) context.Context {
+	return context.WithValue(ctx, traceIDKey, traceID)
+}
+
+// CtxGetTraceID gets the trace ID associated with this context.
+// Return the trace ID and true if there is a trace ID, empty and false otherwise.
+func CtxGetTraceID(ctx context.Context) (string, bool) {
+	return ctxget(ctx, traceIDKey)
+}
+
+// CtxWithOrgID creates a new [context.Context] with the given organization ID associated with it.
+// Call [CtxGetOrgID] to retrieve the organization ID.
+func CtxWithOrgID(ctx context.Context, orgID string) context.Context {
+	return context.WithValue(ctx, orgIDKey, orgID)
+}
+
+// CtxGetOrgID gets the trace ID associated with this context.
+// Return the trace ID and true if there is a trace ID, empty and false otherwise.
+func CtxGetOrgID(ctx context.Context) (string, bool) {
```

Comment:
I have mixed feelings about the bool. I'd say that is unlikely to return false, and we should try to make sure that the fields are available. Regarding usage can make a little harder to use as you cannot inline it.

Reply:
Also, zero value for string would be perfect fine to check if we need

Reply:
I usually like the semantics from maps where I can always be sure if the zero value was inserted there or if it is an actual default zero value, but for maps you can always also ignore the boolean when you dont want (Go cheats =P...it can do cool stuff you cant). But in this case it does seem that inserting an empty string would be something very odd anyway (getenv has similar behavior and an alternative that returns a boolean, because sometimes it makes a difference if it was exported with empty or undefined etc).

Reply:
Yep, having boolean being optional would be a nice thing. I also like the map as it's your decision if the zero value is relevant to your context or not.